nbbuild/antsrc/org/netbeans/nbbuild/CustomJavac.java

```diff
@@ -252,4 +276,144 @@ private static boolean isIgnored(
+            if (c.getClassLoader() != loader) {
+                throw new BuildException("Class " + c + " loaded by " + c.getClassLoader() + " and not " + loader);
+            }
+            Class<?> stdLoc = c.getClassLoader().loadClass(StandardLocation.class.getName());
+            if (stdLoc.getClassLoader() != loader) {
+                throw new BuildException("Class " + stdLoc + " loaded by " + stdLoc.getClassLoader() + " and not " + loader);
+            }
+        }
+
+        @Override
+        protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+            if (isNbJavacClass(name)) {
+                Class<?> clazz = priorityLoaded.get(name);
+                if (clazz == null) {
+                    clazz = findClass(name);
+                    priorityLoaded.put(name, clazz);
+                }
+                return clazz;
+            }
+            return super.loadClass(name, resolve);
+        }
+
+        @Override
+        public Class<?> loadClass(String name) throws ClassNotFoundException {
+            return loadClass(name, false);
+        }
+
+        private static boolean isNbJavacClass(String name) {
+            return name.startsWith("javax.annotation.")
+                    || name.startsWith("javax.tools.")
+                    || name.startsWith("javax.lang.model.")
+                    || name.startsWith("com.sun.source.")
+                    || name.startsWith("com.sun.tools.");
+        }
+    }
+
+    private static final class NbJavacCompiler extends Javac13 {
+
+        private final Class<?> mainClazz;
+
+        NbJavacCompiler(Class<?> mainClazz) {
+            this.mainClazz = mainClazz;
+        }
+
+        @Override
+        public boolean execute() throws BuildException {
+            attributes.log("Using modern compiler", Project.MSG_VERBOSE);
+            Commandline cmd = setupModernJavacCommand();
+            final String[] args = cmd.getArguments();
+            boolean bootClasspath = false;
+            for (int i = 0; i < args.length; i++) {
+                if (args[i].startsWith("-Xbootclasspath/p:")) { // ide/html
+                    bootClasspath = true;
+                }
+                if (args[i].startsWith("-J")) {
+                    args[i] = "-Xlint:none"; // webcommon/javascript2.editor
+                }
+            }
+            for (int i = 0; i < args.length; i++) {
+                if (!bootClasspath) {
+                    if ("-target".equals(args[i]) || "-source".equals(args[i])) {
+                        args[i] = "--release";
+                        if (args[i + 1].startsWith("1.")) {
+                            args[i + 1] = "8";
+                        }
+                    }
+                }
+                if ("-Werror".equals(args[i])) {
+                    args[i] = "-Xlint:none";
```

Comment:
If I understand this correctly this turn a "fail build on error" into an "suppress errors". This sounds strange to me.

Reply:
If I run with `-Werror`, then the build immediately fails on [these annotation warnings](https://github.com/apache/netbeans/actions/runs/4459083199/jobs/7831316138?pr=5609#step:5:54988) caused by the way I prepare `ct.sym` for [nb-javac](https://github.com/JaroslavTulach/nb-javac). The problems can be fixed in some new version of [nb-javac](https://github.com/JaroslavTulach/nb-javac), but for version 20.0.0 the removal of `-Werror` is necessary.

Btw. this shows why the _"dog food testing"_ of [nb-javac](https://github.com/JaroslavTulach/nb-javac) on NetBeans sources makes sense. These warnings are present even when we run [nb-javac](https://github.com/JaroslavTulach/nb-javac) internally for Java editor purposes. They are just not visible (much) and we would never feel the need to address them.

I'll see what I can do to eliminate the [annotation warnings](https://github.com/apache/netbeans/actions/runs/4459083199/jobs/7831316138?pr=5609#step:5:54988) in a future version of [nb-javac](https://github.com/JaroslavTulach/nb-javac).

Reply:
Mitigated greatly by Matthias's 42ed77a